src/main/frontend/src/components/itempage/ItemInfo.js

```diff
@@ -0,0 +1,128 @@
+import classes from "./ItemInfo.module.css";
+import { Formik, Form, Field } from "formik";
+import { Button } from "react-bootstrap";
+import { AuthContext } from "../../hooks";
+import { useContext } from "react";
+import { itemBid } from "../../utils/itemService";
+import { useEffect, useState } from "react";
+import ItemDetails from "./ItemDetails";
+import { computeTimeLeft } from "../../utils/itemUtils";
+
+function ItemInfo({
+  bids,
+  auctionEndDate,
+  itemId,
+  name,
+  startingPrice,
+  details,
+}) {
+  const { token, isUserLoggedIn } = useContext(AuthContext);
+  const [bidResponse, setBidResponse] = useState();
+  const [currentAmount, setCurrentAmount] = useState(0);
+  const [currentNumberOfBids, setCurrentNumberOfBids] = useState(0);
+  const [timeLeft, setTimeLeft] = useState(0);
+
+  useEffect(async () => {
+    if (bids) {
+      setCurrentNumberOfBids(bids.length);
+      setCurrentAmount(getHighestBid());
+
+      let endDate = auctionEndDate.slice(0, 10).split("-");
+      let endTime = auctionEndDate.slice(11).split(":");
+      endDate = new Date(
+        endDate[0],
+        endDate[1] - 1,
+        endDate[2],
+        endTime[0],
+        endTime[1],
+        endTime[2]
+      );
+
+      setTimeLeft(computeTimeLeft(endDate));
+    }
+  }, [bids]);
+
+  const handleSubmit = async (item) => {
+    const { amount } = item;
+    if (amount <= currentAmount) {
+      setBidResponse(`Minimal amount required is ${currentAmount + 1}`);
+    } else {
+      try {
+        const itemRes = await itemBid(token, {
+          itemId,
+          amount: parseFloat(amount),
+        });
+        setBidResponse(itemRes.body);
+        setCurrentAmount(amount);
+        setCurrentNumberOfBids(currentNumberOfBids + 1);
+      } catch (e) {
+        console.error(e);
+      }
+    }
+  };
+
+  function getHighestBid() {
```

Comment:
do not mix es6 and es5 type of defining functions. choose one and use it in whole project, or at least make it consistent on file basis